addons/external-dns/gcp/main.tf

```diff
@@ -0,0 +1,26 @@
+data "terraform_remote_state" "infra" {
+  backend = "local"
+  config = {
+    path = "../../../infra/gcp/terraform.tfstate.d/gcp-${var.cluster_id}-${var.gcp_k8s_regions[var.cluster_id]}/terraform.tfstate"
+  }
+}
+
+data "terraform_remote_state" "dns" {
+  backend = "local"
+  config = {
+    path = "credentials/terraform.tfstate"
```

Comment:
we should limit the referencing of the states across the modules and do communicate the variables over.

Reply:
Yes, we should, but the credentials are only returned _on creation_. You can't get the private key with a data provider and you can't get it after wards.
Since we only configure DNS in one project, not once per cluster, we don't have any other option than storing the credential locally so that we can reference it for all clusters.

Reply:
so anyway this is a function of "output" for the given module, account we follow `module/gcp/external-dns ` comment